test/e2e/gateway_api_test.go

```diff
@@ -0,0 +1,189 @@
+	t.Helper()
+	gatewayFailed, gatewayClassFailed, httpRouteFailed = false, false, false
+
+	// Check if GatewayClass can be created.
+	gatewayClass = buildGatewayClass("openshift-default", "openshift.io/gateway-controller")
+	if err := kclient.Create(context.TODO(), &gatewayClass); err != nil {
+		gatewayClassFailed = true
+		if log {
+			t.Logf("error creating gateway class: %v", err)
+		}
+	}
+
+	// Check if Gateway can be created.
+	gateway = buildGateway("test-gateway", "openshift-ingress", gatewayClass.Name)
+	if err := kclient.Create(context.TODO(), &gateway); err != nil {
+		gatewayFailed = true
+		if log {
+			t.Logf("error creating gateway: %v", err)
+		}
+	}
+
+	// Check if HTTPRoute can be created.
+	httpRoute := buildHTTPRoute("test-httproute", ns, gateway.Name, "openshift-ingress", "test-hostname", "test-app", 8080)
+	if err := kclient.Create(context.TODO(), httpRoute); err != nil {
+		httpRouteFailed = true
+		if log {
+			t.Logf("error creating http route: %v", err)
+		}
+	}
+
+	return gatewayClassFailed, gatewayFailed, httpRouteFailed
+}
+
+func buildGatewayClass(name, controllerName string) v1beta1.GatewayClass {
+	return v1beta1.GatewayClass{
+		ObjectMeta: metav1.ObjectMeta{Name: name},
+		Spec: v1beta1.GatewayClassSpec{
+			ControllerName: v1beta1.GatewayController(controllerName),
+		},
+	}
+}
+
+func buildGateway(name, namespace, gcname string) v1beta1.Gateway {
+	hostname := v1beta1.Hostname(testHostname)
+	listener1 := v1beta1.Listener{Name: "http", Hostname: &hostname, Port: 80, Protocol: "HTTP"}
+	return v1beta1.Gateway{
+		ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: namespace},
+		Spec: v1beta1.GatewaySpec{
+			GatewayClassName: v1beta1.ObjectName(gcname),
+			Listeners:        []v1beta1.Listener{listener1},
+		},
+	}
+}
+
+func buildHTTPRoute(routename, namespace, parentgateway, parentnamespace, hostname, backendrefname string, backendrefport int) *v1beta1.HTTPRoute {
+	parentns := v1beta1.Namespace(parentnamespace)
+	parent := v1beta1.ParentReference{Name: v1beta1.ObjectName(parentgateway), Namespace: &parentns}
+	port := v1beta1.PortNumber(backendrefport)
+	rule := v1beta1.HTTPRouteRule{
+		BackendRefs: []v1beta1.HTTPBackendRef{{
+			BackendRef: v1beta1.BackendRef{
+				BackendObjectReference: v1beta1.BackendObjectReference{
+					Name: v1beta1.ObjectName(backendrefname),
+					Port: &port,
+				},
+			},
+		}},
+	}
+
+	return &v1beta1.HTTPRoute{
+		ObjectMeta: metav1.ObjectMeta{Name: routename, Namespace: namespace},
+		Spec: v1beta1.HTTPRouteSpec{
+			CommonRouteSpec: v1beta1.CommonRouteSpec{ParentRefs: []v1beta1.ParentReference{parent}},
+			Hostnames:       []v1beta1.Hostname{v1beta1.Hostname(hostname)},
+			Rules:           []v1beta1.HTTPRouteRule{rule},
+		},
+	}
+}
+
+func buildNamespace(name string) *corev1.Namespace {
+	return &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}
+}
```

Comment:
This helper is likely to be useful in other tests, so it could make sense to put it in `util_test.go`.

Reply:
Removed, no longer used.